spec/controllers/openid_connect/authorizations_controller_spec.rb

```diff
@@ -0,0 +1,172 @@
+               client_id: client_with_multiple_redirects.client_id,
+               response_type: "id_token",
+               scope: "openid",
+               nonce: SecureRandom.hex(16),
+               state: SecureRandom.hex(16)
+             }
+        expect(response.body).to match("bad_request")
+      end
+    end
+    context "when redirect URI does not match pre-registered URIs" do
+      it "should return an invalid request error" do
+        post :new,
+             {
+               client_id: client.client_id,
+               redirect_uri: "http://localhost:2000/",
+               response_type: "id_token",
+               scope: "openid",
+               nonce: SecureRandom.hex(16)
+             }
+        expect(response.body).to match("bad_request")
+      end
+    end
+    context "when an unsupported scope is passed in" do
+      it "should return an invalid scope error" do
+        post :new,
+             {
+               client_id: client.client_id,
+               redirect_uri: "http://localhost:3000/",
+               response_type: "id_token",
+               scope: "random",
+               nonce: SecureRandom.hex(16),
+               state: SecureRandom.hex(16)
+             }
+        expect(response.body).to match("error=invalid_scope")
+      end
+    end
+    context "when nonce is missing" do
+      it "should return an invalid request error" do
+        post :new,
+             {
+               client_id: client.client_id,
+               redirect_uri: "http://localhost:3000/",
+               response_type: "id_token",
+               scope: "openid",
+               state: SecureRandom.hex(16)
+             }
+        expect(response.location).to match("error=invalid_request")
+      end
+    end
+  end
+
+  describe "#create" do
+    before do
+      get :new,
+          {
+            client_id: client.client_id,
+            redirect_uri: "http://localhost:3000/",
+            response_type: "id_token",
+            scope: "openid",
+            nonce: SecureRandom.hex(16),
+            state: 4180930983
+          }
+    end
+    context "when authorization is approved" do
+      before do
+        post :create,
+             {
+               approve: "true"
+             }
+      end
+      it "should return the id token in a fragment" do
+        expect(response.location).to have_content("id_token=")
+      end
+      it "should return the passed in state" do
+        expect(response.location).to have_content("state=4180930983")
+      end
+    end
+    context "when authorization is denied" do
+      before do
+        post :create,
+             {
+               approve: "false"
+             }
+      end
+      it "should return an error in the fragment" do
+        expect(response.location).to have_content("error=")
+      end
+      it "should NOT contain a id token in the fragment" do
+        expect(response.location).to_not have_content("id_token=")
+      end
+    end
+  end
+
+end
```

Comment:
I think it would be good to maintain some blank lines between the examples. That is an `end` should be followed by either another `end` or an empty line.